pkg/frontend/adminactions/metricsactions.go

```diff
@@ -0,0 +1,69 @@
+package adminactions
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	metricsclient "k8s.io/metrics/pkg/client/clientset/versioned"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/env"
+	"github.com/Azure/ARO-RP/pkg/util/restconfig"
+)
+
+// MetricActions are those that involve k8s objects, and thus depend upon k8s clients being createable
+type MetricActions interface {
```

Comment:
We already have `KubeActions` for actions where we call Kube API.

Reply:
I separated this out because PodMetrics and NodeMetrics are not available through the `clientset` being used in `KubeActions`.

MetricActions - [k8s.io/metrics clientset](https://github.com/kubernetes/metrics/blob/master/pkg/client/clientset/versioned/clientset.go)
KubeActions - [openshift/client-go clientset](https://github.com/openshift/client-go/blob/master/config/clientset/versioned/clientset.go)

Note that the `KubeActions` clientset is pulled from the OpenShift repo, while the `MetricAction` clientset I pulled in is from the k8s upstream metrics repo. There is an [openshift/kubernetes-metrics](https://github.com/openshift/kubernetes-metrics/blob/master/pkg/client/clientset/versioned/clientset.go) repo that may be able to do the same thing, but these files seem like they haven't been touched in quite some time.

All of this said, there is no reason that `MetricActions` and `KubeActions` could not be merged together and the individual functions use the appropriate clientsets. I just wanted to make the difference more obvious.

Thoughts?

Reply:
1. KubeActions use dynamic client `k8s.io/client-go/dynamic`. I believe should be able to work with any resource. We might be able to do the same thing without using `k8s.io/metrics` clientset.
1. KubeActions can be extened with additional clients, if needed.

I'm not 100% sure that we should be combining these two things under one interface, but we should consider if we decide to proceed with introducing this functionality.

Reply:
> KubeActions use dynamic client `k8s.io/client-go/dynamic`. I believe should be able to work with any resource. We might be able to do the same thing without using `k8s.io/metrics` clientset.

@m1kola We can test this by running the existing "List or get Kubernetes objects" Geneva Action for `PodMetrics` and `NodeMetrics`. I have an ask out to a few of the Red Hat and MS team members to try this, as I do not have access to run Geneva Actions. Is this something that you could test as well?